JarrydWorland/2021_Capstone_GP2_Team
# Refactor level traversal

## Changes include
- Added wall colliders (dc30465)
- Added CameraPanBehaviour (5c0aea7) 
- Set door colliders to trigger (1cd61db)
- Added ToVector3 extension method to Direction (918ad9a)
- Implemented level traversal (553e4eb)
    - Implemented LevelTraversalBehaviour
    - Added PlayerLevelTraversalBehaviour

This PR refactors the players ability to travel in-between rooms through doors, prevents the player from walking through the walls of the rooms and updates the camera to keep track of the room the player is currently in. 

## How to test
Place a level and a player, and attach a CameraPanBehaviour to the default camera. The player should be able to move in-between rooms by traveling through the doors. The speed of the camera pan can be adjusted on the CameraPanBehaviour component. It is slow by default and a better default will be set when a proper scene is added.
